src/components/ProgramRecord/ProgramRecordTable.jsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { DataTable, Badge } from '@edx/paragon';
+import { FormattedMessage } from '@edx/frontend-platform/i18n';
+
+const ProgramRecordTable = ({ grades }) => {
+  const formatGradeData = (data) => (
+    data.map(grade => ({
+      ...grade,
+      course_id: grade.course_id.replace(/^course-v1:/, ''),
+      // If certificate not earned hide some fields
+      ...(!grade.issue_date && { course_id: null }),
+      ...(!grade.issue_date && { letter_grade: null }),
+      ...(!grade.issue_date && { attempts: null }),
+      percent_grade:
+        grade.issue_date
+          ? `${parseInt(Math.round(grade.percent_grade * 100), 10).toString()}%`
+          : '',
+      issue_date:
+        grade.issue_date
+          ? new Date(grade.issue_date).toLocaleDateString()
+          : '',
+      status: grade.issue_date
+        ? (
+          <Badge variant="success">
+            <FormattedMessage
+              id="earned.course.credential"
+              defaultMessage="Earned"
+              description="Course credential that has been earned"
+            />
+          </Badge>
+        )
+        : (
+          <Badge variant="danger">
+            <FormattedMessage
+              id="not.earned.course.credential"
+              defaultMessage="Not Earned"
+              description="Course credential that has not been earned"
+            />
```

Comment:
It looks like these strings (`Earned` and `Not Earned`) are used both in the `ProgramRecordTable` and the `ProgramRecordHeader` components. Can we re-use this? That way the string only has to be extracted-and-translated once? 

It's common practice for us to have separate "messages" files that we can import. There should be examples in the Communications MFE you can follow.